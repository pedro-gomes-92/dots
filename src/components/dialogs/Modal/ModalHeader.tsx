import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { CardHeader, CardHeaderProps } from '../../containers';
import { Close } from '../../miscellaneous';
import { BaseDialogProps } from '../BaseDialog';

export interface ModalHeaderProps extends Omit<CardHeaderProps, 'actions'> {
  onClose?: BaseDialogProps['onClose'];
}

export const ModalHeader: FunctionComponent<ModalHeaderProps> = ({
  className,
  onClose,
  ...rest
}: ModalHeaderProps): JSX.Element => {
  const actions = [];
  if (onClose) {
    actions.push(<Close key="key-close-dialog" onClick={onClose} />);
  }

  return <CardHeader actions={<>{actions}</>} className={classnames('modal-header', className)} {...rest} />;
};

ModalHeader.defaultProps = {
  onClose: (): void => {},
};
