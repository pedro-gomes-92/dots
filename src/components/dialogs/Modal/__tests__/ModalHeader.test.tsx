import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { ModalHeader } from '../ModalHeader';

describe('Given <ModalHeader />', (): void => {
  let container: HTMLElement;

  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      container = render(<ModalHeader />).container;
      expect(container).toMatchSnapshot();
    });

    describe('And uses onClose', (): void => {
      let handleClose: jest.Mock;
      let close: HTMLElement;

      beforeEach((): void => {
        handleClose = jest.fn();
        container = render(<ModalHeader onClose={handleClose} />).container;
        close = container.querySelector('.close');

        fireEvent.click(close);
      });

      it('Then should match the snapshot', (): void => {
        expect(container).toMatchSnapshot();
      });

      it('And should render close button', (): void => {
        expect(close).toBeTruthy();
      });

      it('And should trigger onClose', (): void => {
        expect(handleClose).toHaveBeenCalledTimes(1);
      });
    });
  });
});
