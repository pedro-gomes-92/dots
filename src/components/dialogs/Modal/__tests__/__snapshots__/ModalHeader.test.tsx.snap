// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <ModalHeader /> When initializes And uses onClose Then should match the snapshot 1`] = `
<div>
  <div
    class="card-header modal-header"
  >
    <div
      class="layout is-2 is-mobile columns"
    >
      <div
        class="layout-item column"
      >
        <h3
          class="is-3 text title"
        />
      </div>
      <div
        class="is-auto layout-item column"
      >
        <button
          class="button is-text close"
        >
          <span
            class="icon"
          >
            <i
              class="icon-font icon-font-close"
            />
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Given <ModalHeader /> When initializes Then should match the snapshot 1`] = `
<div>
  <div
    class="card-header modal-header"
  >
    <div
      class="layout is-2 is-mobile columns"
    >
      <div
        class="layout-item column"
      >
        <h3
          class="is-3 text title"
        />
      </div>
      <div
        class="is-auto layout-item column"
      >
        <button
          class="button is-text close"
        >
          <span
            class="icon"
          >
            <i
              class="icon-font icon-font-close"
            />
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
